R/meta_regression/regression_plot_panel.R

```diff
@@ -0,0 +1,276 @@
+#'
+#' @param id ID of the module.
+#' @return Div containing the module UI.
+regression_plot_panel_ui <- function(id) {
+  ns <- NS(id)
+  
+  sidebarLayout(
+    position = "right",
+    mainPanel = mainPanel(
+      width = 9,
+      plotOutput(outputId = ns("regression_plot"), height = "800px")
+    ),
+    sidebarPanel = sidebarPanel(
+      width = 3,
+      # Add comparators
+      selectInput(inputId = ns("add_comparator_dropdown"), label = "Add Comparator", choices = c(), selectize = FALSE),
+      actionButton(inputId = ns("add_comparator_btn"), label = "Add to plot"),
+      div(
+        style = "float: right;",
+        actionButton(inputId = ns("add_all_btn"), label = "Add all")
+      ),
+      br(),
+      br(),
+      # Remove comparators
+      selectInput(inputId = ns("remove_comparator_dropdown"), label = "Remove Comparator", choices = c(), selectize = FALSE),
+      actionButton(inputId = ns("remove_comparator_btn"), label = "Remove"),
+      div(
+        style = "float: right;",
+        actionButton(inputId = ns("remove_all_btn"), label = "Remove all")
+      ),
+      
+      h3("Plot Options"),
+      # Covariate value
+      .AddTooltip(
+        checkboxInput(inputId = ns("covariate"), label = "Show covariate value", value = TRUE),
+        tooltip = "Show the covariate value as a vertical line at the current value"
+      ),
+      # Confidence regions
+      .AddTooltip(
+        checkboxInput(inputId = ns("confidence"), label = "Show confidence regions"),
+        tooltip = "Show confidence regions for the added comparisons"
+      ),
+      sliderInput(
+        inputId = ns("confidence_opacity"),
+        label = "Confidence Region Opacity",
+        min = 0,
+        max = 0.5,
+        step = 0.01,
+        value = 0.2
+      ),
+      # Regression lines
+      .AddTooltip(
+        checkboxInput(inputId = ns("ghosts"), label = "Show ghost comparisons", value = TRUE),
+        tooltip = "Show other available comparisons in light grey behind the main plot"
+      ),
+      .AddTooltip(
+        checkboxInput(inputId = ns("extrapolate"), label = "Extrapolate regression lines", value = TRUE),
+        tooltip = "Extrapolate the regression lines beyond the range of the original data"
+      ),
+      # Contributions
+      .AddTooltip(
+        checkboxInput(inputId = ns("contributions"), label = "Show contributions", value = TRUE),
+        tooltip = "Show study contributions as circles, where a bigger circle represents a larger contribution"
+      ),
+      
+      div(
+        id = ns("contribution_options"),
+        p("Study circle sized by:"),
+        div(
+          .CreateInlineBlock("% Contribution", style = "padding-right: 10pt;"),
+          shinyWidgets::materialSwitch(
+            inputId = ns("contribution_toggle"),
+            inline = TRUE,
+            status = "primary"
+          ),
+          .CreateInlineBlock("Inverse Variance")
+        ),
+        div(
+          .CreateInlineBlock(
```

Comment:
I like the disabling of the widgets but here, where there are two linked to the "show contributions" tick box it is less obvious that the circle size multiplier is linked

Reply:
It should be more obvious now that all the disabling is fixed.